### PR TITLE
Change encoding passed to StreamWriter constructor

### LINKED ISSUE #85: TextSerializationWriter should not use System.Text.Encoding.UTF8
In #83, a change was made to the `MemoryStream` used by the TextSerializationWriter to leave the underlying `Stream` open. Specifically, the change was to pass the `leaveOpen` parameter as true.

This was done for .NET like so:

```csharp
writer = new(_stream, leaveOpen: true);
```

This skipped the two optional parameters between the stream and `leaveOpen`: `encoding` and `bufferSize`, so that the default values of these parameters would be used.

However, the testing section of the "Build and Test" action runs tests under the .NET Framework 4.6.2, which doesn't compile if you skip optional parameters. The attempt to address this was to change to:

```csharp
writer = new(
    _stream,
    // Default encoding
    encoding: System.Text.Encoding.UTF8,
    // Default buffer size
    bufferSize: 1024,
    leaveOpen: true);
```

However, it turns out that default encoding actually is not `System.Text.Encoding.UTF8`. It's a custom encoding:

https://referencesource.microsoft.com/#mscorlib/system/io/streamwriter.cs,106

```csharp
UTF8Encoding noBOM = new UTF8Encoding(false, true);
```

### PR DESCRIPTION
**Pull request by jasonjoh**
Fixes #85

### CODE REVIEW

File: src/TextSerializationWriter.cs [hunk 1]
```diff
@@ -23,7 +23,7 @@ public TextSerializationWriter()
         writer = new(
             _stream,
             // Default encoding
-            encoding: System.Text.Encoding.UTF8,
+            encoding: new System.Text.UTF8Encoding(false, true),
```
Comment (baywet) [inline src/TextSerializationWriter.cs:26]:
```suggestion
```

File: src/TextSerializationWriter.cs [hunk 2]
```diff
@@ -23,7 +23,7 @@ public TextSerializationWriter()
...
             // Default buffer size
             bufferSize: 1024,
```
Comment (baywet) [inline src/TextSerializationWriter.cs:28]:
```suggestion
```

File: src/TextSerializationWriter.cs (hunk 1 above)
Comment (jasonjoh) [reply src/TextSerializationWriter.cs:26]: It doesn't build that way. I think netstandard2.0 and netstandard2.1 don't support named params?

```
C:\Source\Repos\kiota-serialization-text-dotnet\src\TextSerializationWriter.cs(23,18): error CS7036: There is no argument given that corresponds to the required parameter 'encoding' of 'StreamWr
iter.StreamWriter(Stream, Encoding, int, bool)' [C:\Source\Repos\kiota-serialization-text-dotnet\src\Microsoft.Kiota.Serialization.Text.csproj::TargetFramework=netstandard2.0]
C:\Source\Repos\kiota-serialization-text-dotnet\src\TextSerializationWriter.cs(23,18): error CS7036: There is no argument given that corresponds to the required parameter 'encoding' of 'StreamWr
iter.StreamWriter(Stream, Encoding, int, bool)' [C:\Source\Repos\kiota-serialization-text-dotnet\src\Microsoft.Kiota.Serialization.Text.csproj::TargetFramework=netstandard2.1]
```

Comment (baywet) [reply src/TextSerializationWriter.cs:26]: thanks for double checking

**Review by baywet — APPROVED** [review]

**Pull request merged**